src/templates/exportService.hbs

```diff
@@ -38,137 +36,118 @@ const {{{this}}}Validator = ajv.compile(${{{this}}});
 {{/if}}
 {{/if}}
 
-{{#each operations}}
 
-type {{{name}}}OkResult = 
-{{#each okResults}}
-  | {{{base}}}
-{{else}}
-  | void
-{{/each}};
+interface IResult {
+    loading: Boolean;
+    error: CommonErrorResponseApplicationJsonResponse | null;
+    data: {{#if imports}}{{#each imports}}{{{this}}}|{{/each}}{{/if}}Number|String|null|undefined|Array<any>;
+}
 
-type {{{name}}}ErrorResult = 
-{{#each errorResults}}
-  | {{{base}}}
-{{else}}
-  | void
-{{/each}};
+interface IActionCallback {
+    (atom: IResult, set: (update: SetStateAction<IResult>) => void, result: ApiResult): void;
+}
 
+{{#each operations}}
 interface I{{{name}}}Controller {
     fetchData: ({{>parameters}}
         options?: ApiRequestOptions,
-        cb?: IActionCallback<
-            {{{name}}}OkResult, 
-            {{{name}}}ErrorResult
-        >,
-        setter?: (
-            update: SetStateAction<ResultState<
-                {{{name}}}OkResult, 
-                {{{name}}}ErrorResult
-            >>
-        ) => void,
-    ) => void;
+        cb?: IActionCallback,
+        setter?: (update: SetStateAction<IResult>) => void,
+    ) => Promise<ApiResult>;
 };
 
-export function {{{hookName}}}(): [
-    ResultState<
-        {{{name}}}OkResult, 
-        {{{name}}}ErrorResult
-    >, 
-    I{{{name}}}Controller
-] {
-    const {{{name}}}ResultAtom = atom<
-        ResultState<
-            {{{name}}}OkResult, 
-            {{{name}}}ErrorResult
-        >
-    >({ loading: false, error: null, data: null });
-    const [resultState, setter] = useAtom({{{name}}}ResultAtom);
-    const fetchData = async ({{>parameters}}
+{{/each}}
+
+{{#each operations}}
+const {{{name}}}ResultAtom = atom<IResult>({ loading: false, error: null, data: null });
+export function {{{hookName}}}():[IResult, I{{{name}}}Controller] {
+    const [resultAtom, setter] = useAtom({{{name}}}ResultAtom);
```

Comment:
The variable name is wrong, it's a state but not an atom, please change the variable name to `resultAtom`.